### PR TITLE
feat: ✨ Re-write all inference rtdetr(v1/v2) models based original repository

### PR DESCRIPTION
**Pull request by onuralpszr**
I re-wrote rtdetr inference via pytorch only and I also added v2 results as well. Pure pytorch runs gives and closer results @LinasKo and I also tested on val2017 data set closer mAP results example case result for rtdetr_r18vd (v1) 

```json

{
    "iou_thresholds": [
        0.5,
        0.55,
        0.6,
        0.65,
        0.7,
        0.75,
        0.8,
        0.85,
        0.8999999999999999,
        0.95
    ],
    "map50_95": 0.4475030676297469,
    "map50": 0.6156237779572045,
    "map75": 0.48407456419504113,
    "small_objects": {
        "map50_95": 0.19035576042172211,
        "map50": 0.3089236487784832,
        "map75": 0.2030844478437464
    },
    "medium_objects": {
        "map50_95": 0.3952415547608772,
        "map50": 0.5612930297373171,
        "map75": 0.442782433290807
    },
    "large_objects": {
        "map50_95": 0.5894997261827915,
        "map50": 0.7265982743927857,
        "map75": 0.6411468239795821
    },
    "metadata": {
        "model": "rtdetr_r18vd",
        "param_count": 21955472,
        "run_date": "2024-09-09T23:25:16.861527+00:00"
    }
}
```

### CODE REVIEW

**Comment by LinasKo** [comment]
To add some context:

The number we had for RT-DETR previously were obtained by pulling RT-DETR from pytroch hub, rather than running it in the same way as the original. The results were suspicious as version with Objects365 dataset pretraining sometimes performed worse than the baseline, which is not expected.

Onuralp tested on the validation set of COCO 2017, as that's what the [original used](https://github.com/lyuwenyu/RT-DETR/blob/main/benchmark/README.md). The evaluation results were similar.

**Review by LinasKo — APPROVED** [review]
